File: Algorithms/Find All Duplicates in an Array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        n=len(nums)
        arr=[0]*(n+1)
        for i in range(n):
            arr[nums[i]]+=1
        result=[]
        for i in range(1,n+1):
            if arr[i]==2:
                result.append(i)
        return result

File: Algorithms(Java)/Find All Duplicates in an Array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        HashSet<Integer> h=new HashSet<Integer>();
        List<Integer> l=new ArrayList<Integer>();
        for (int i=0;i<nums.length;i++){
            if (h.contains(nums[i])){
                l.add(nums[i]);
            }
            h.add(nums[i]);
        }
        return l;
    }
}